File: Android/app/src/main/java/sr57/ftn/redditclone/service/CommunityService.java
package sr57.ftn.redditclone.service;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;
import sr57.ftn.redditclone.model.Community;
import sr57.ftn.redditclone.model.DTO.AddCommunityDTO;
import sr57.ftn.redditclone.model.DTO.AddPostDTO;
import sr57.ftn.redditclone.model.Post;

public interface CommunityService {

    @Headers({
            "User-Agent: Mobile-Android",
            "Content-Type:application/json"
    })

    @GET("api/communities/all")
    Call<List<Community>> getAll();

    @GET("api/communities/{community_id}")
    Call<Community> getSelectedCommunityById(@Path("community_id") int community_id);

    @GET("api/communities/{community_id}/posts")
    Call<List<Post>> getSelectedCommunityPosts(@Path("community_id") int community_id);

    @POST("api/communities/addCommunityAndroid")
    Call<AddCommunityDTO> addCommunityAndroid(@Body AddCommunityDTO addCommunityDTO);

    @POST("api/communities/{community_id}/addPost")
    Call<AddPostDTO> addPost(@Path("community_id") int community_id, @Body AddPostDTO addPostDTO);

    @PUT("api/communities/updateCommunity/{community_id}")
    Call<Community> updateCommunity(@Body Community community, @Path("community_id") int community_id);

    @PUT("api/communities/suspendCommunity/{community_id}")
    Call<Community> suspendCommunity(@Body Community community, @Path("community_id") int community_id);
}

File: Frontend/src/app/service/community.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Community } from '../model/community.model';
import { Post } from '../model/post.model';

@Injectable({
  providedIn: 'root'
})
export class CommunityService {
  api = "http://localhost:8080/api/communities/"

  constructor(private http: HttpClient) { }

  getAll(): Observable<Array<Community>> {
    return this.http.get<Array<Community>>(this.api + "all");
  }

  addCommunity(community: Community): Observable<Community> {
    return this.http.post<Community>(this.api + "add", community);
  }

  getCommunity(community_id: number): Observable<Community> {
    return this.http.get<Community>(this.api + community_id);
  }

  suspendCommunity(community_id: number, community: Community) {
    return this.http.put(this.api + "suspendCommunity" + "/" + community_id, community);
  }

  updateCommunity(community_id: number, community: Community) {
    return this.http.put(this.api + "updateCommunity" + "/" + community_id, community);
  }

  getCommunityPosts(community_id: number) {
    return this.http.get(this.api + community_id + "/posts")
  }

  getCommunityRules(community_id: number) {
    return this.http.get(this.api + community_id + "/rules")
  }

  getCommunityFlairs(community_id: number) {
    return this.http.get(this.api + community_id + '/flairs')
  }

  addPost(community_id: number, post: Post): Observable<Post> {
    return this.http.post<Post>(this.api + community_id + "/addPost", post);
  }

}
